src/scraper/scraper.cpp

```diff
@@ -1245,131 +1264,383 @@ bool UserpassPopulated()
+* Project Host Files  *
 **********************/
 
-bool DownloadProjectTeamFiles(const NN::WhitelistSnapshot& projectWhitelist)
+bool DownloadProjectHostFiles(const NN::WhitelistSnapshot& projectWhitelist)
 {
-        if (!projectWhitelist.Populated())
+    // If fExplorer is false then skip processing. (This should not be called anyway, but return immediately just in case.
+    if (!fExplorer)
+    {
+        _log(logattribute::INFO, "DownloadProjectHostFiles", "Not in explorer mode. Skipping host file download and processing.");
+        return false;
+    }
+
+    if (!projectWhitelist.Populated())
+    {
+        _log(logattribute::CRITICAL, "DownloadProjectHostFiles", "Whitelist is not populated");
+
+        return false;
+    }
+
+    _log(logattribute::INFO, "DownloadProjectHostFiles", "Whitelist is populated; Contains " + std::to_string(projectWhitelist.size()) + " projects");
+
+    if (!UserpassPopulated())
+    {
+        _log(logattribute::CRITICAL, "DownloadProjectHostFiles", "Userpass is not populated");
+
+        return false;
+    }
+
+    for (const auto& prjs : projectWhitelist)
+    {
+        _log(logattribute::INFO, "DownloadProjectHostFiles", "Downloading project host file for " + prjs.m_name);
+
+        // Grab ETag of host file
+        Http http;
+        std::string sHostETag;
+
+        bool buserpass = false;
+        std::string userpass;
+
+        for (const auto& up : vuserpass)
         {
-            _log(logattribute::CRITICAL, "DownloadProjectTeamFiles", "Whitelist is not populated");
+            if (up.first == prjs.m_name)
+            {
+                buserpass = true;
 
-            return false;
+                userpass = up.second;
+
+                break;
+            }
         }
 
-        _log(logattribute::INFO, "DownloadProjectTeamFiles", "Whitelist is populated; Contains " + std::to_string(projectWhitelist.size()) + " projects");
+        try
+        {
+            sHostETag = http.GetEtag(prjs.StatsUrl("host"), userpass);
+        }
+        catch (const std::runtime_error& e)
+        {
+            _log(logattribute::ERR, "DownloadProjectHostFiles", "Failed to pull host header file for " + prjs.m_name);
+            continue;
+        }
 
-        if (!UserpassPopulated())
+        if (sHostETag.empty())
         {
-            _log(logattribute::CRITICAL, "DownloadProjectTeamFiles", "Userpass is not populated");
+            _log(logattribute::ERR, "DownloadProjectHostFiles", "ETag for project is empty" + prjs.m_name);
 
-            return false;
+            continue;
         }
+        else
+            _log(logattribute::INFO, "DownloadProjectHostFiles", "Successfully pulled host header file for " + prjs.m_name);
 
-        for (const auto& prjs : projectWhitelist)
+        if (buserpass)
         {
-            LOCK(cs_TeamIDMap);
-            if (fDebug3) _log(logattribute::INFO, "LOCK", "cs_TeamIDMap");
+            authdata ad(lowercase(prjs.m_name));
```

Comment:
Scraper v1 question: I notice we're saving the etag to disk--do we do anything with it after that? I can't find where that file is used.

Reply:
No. And that temp file is deleted, plus the etag used is in the filenames of the stats. We could probably just keep it in memory. I'll take a look at that for Part II.

Reply:
Okie doke. If we do end up using it for something, we'll need to remember to go back and make the project name casing match.

Reply:
True. Not worth worrying about it right now though I think.